*** build/tensorflow_src/tensorflow/lite/simple_memory_arena_debug_dump.cc.org	Sat Nov 20 16:03:49 2021
--- build/tensorflow_src/tensorflow/lite/simple_memory_arena_debug_dump.cc	Sat Nov 20 16:04:08 2021
***************
*** 18,23 ****
--- 18,24 ----
  #include <queue>
  #include <string>
  #include <vector>
+ #include <limits>
  
  #include "tensorflow/lite/simple_memory_arena.h"
  
*** build/cpuinfo-source/src/x86/windows/init.c.org	Fri Jul 30 05:21:33 2021
--- build/cpuinfo-source/src/x86/windows/init.c	Fri Jul 30 05:21:58 2021
***************
*** 10,15 ****
--- 10,21 ----
  
  #include <windows.h>
  
+ #ifndef max
+ static inline uint32_t max(uint32_t a, uint32_t b) {
+ 	return a > b ? a : b;
+ }
+ #endif
+ 
  #ifdef __GNUC__
    #define CPUINFO_ALLOCA __builtin_alloca
  #else
